gcloud-java-resourcemanager/src/main/java/com/google/gcloud/resourcemanager/Policy.java

```diff
@@ -0,0 +1,389 @@
+            case "serviceAccount":
+              members.add(new Member(MemberType.SERVICE_ACCOUNT, checkNotNull(emailOrDomain)));
+              break;
+            case "group":
+              members.add(new Member(MemberType.GROUP, checkNotNull(emailOrDomain)));
+              break;
+            case "domain":
+              members.add(new Member(MemberType.DOMAIN, checkNotNull(emailOrDomain)));
+              break;
+            default:
+              throw new UnsupportedOperationException("Unsupported member type: " + memberTypeStr);
+          }
+        }
+      }
+      return new Binding(role, members);
+    }
+
+    @Override
+    public int hashCode() {
+      return Objects.hash(role, members);
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+      return obj instanceof Binding && Objects.equals(this.role, ((Binding) obj).role)
+          && Objects.equals(this.members, ((Binding) obj).members);
+    }
+  }
+
+  public static final class Builder {
+    private List<Binding> bindings;
+    private String etag;
+    private Integer version;
+
+    private Builder() {
+      bindings = new ArrayList<>();
+    }
+
+    public Builder addBinding(Binding binding) {
+      this.bindings.add(binding);
+      return this;
+    }
+
+    public Builder removeBinding(Binding binding) {
+      this.bindings.remove(binding);
+      return this;
+    }
+
+    public Builder clearBindings() {
+      this.bindings = new ArrayList<>();
+      return this;
+    }
+
+    public Builder bindings(List<Binding> bindings) {
+      this.bindings = checkNotNull(bindings);
+      return this;
+    }
+
+    public Builder etag(String etag) {
+      this.etag = etag;
+      return this;
+    }
+
+    public Builder version(Integer version) {
+      this.version = version;
+      return this;
+    }
+
+    public Policy build() {
+      return new Policy(this);
+    }
+  }
+
+  Policy(Builder builder) {
+    ImmutableList.Builder<Binding> bindingsListBuilder = new ImmutableList.Builder<>();
+    for (Binding binding : builder.bindings) {
+      bindingsListBuilder.add(binding);
+    }
+    bindings = bindingsListBuilder.build();
+    version = builder.version;
+    etag = builder.etag;
+  }
+
+  public List<Binding> bindings() {
+    return bindings;
+  }
+
+  public Integer version() {
+    return version;
+  }
+
+  public String etag() {
+    return etag;
+  }
+
+  public static Builder builder() {
+    return new Builder();
+  }
+
+  public Builder toBuilder() {
+    List<Binding> mutableBindings = new ArrayList<>();
+    for (Binding binding : bindings) {
+      mutableBindings.add(binding);
+    }
+    return new Builder().bindings(mutableBindings).etag(etag).version(version);
+  }
+
+  @Override
+  public int hashCode() {
+    return Objects.hash(bindings, etag, version);
+  }
+
+  @Override
+  public boolean equals(Object obj) {
+    return obj instanceof Policy && Objects.equals(this.bindings, ((Policy) obj).bindings)
```

Comment:
This may be a silly question, but would that only work if the `com.google.api.services.cloudresourcemanager.model.Policy` class overrides the equals method?  Because the `AbstractMap